### PR TITLE
Ethan - Replaced ReviewTable placeholder pages with ones that includes ReviewTable

### LINKED ISSUE #20: Replace ReviewTable placeholder page with one that includes Review Table

### PR DESCRIPTION
**Pull request by ethanlee556**
Close #20 
Removed placeholder pages/test and replaced them with working pages/test for Review Create, Edit, and Index
image

### CODE REVIEW

**Comment by tallyhawley** [comment]
https://f22-5pm-1-team03-qa.herokuapp.com/
When trying to access your IndexPage, it just... breaks. And requires a reload for any page on the site to function. Looking at the console, it's a type error of some kind. 

image

If you could fix this I'll merge it because it looks good otherwise

**Comment by ethanlee556** [comment]
> https://f22-5pm-1-team03-qa.herokuapp.com/ When trying to access your IndexPage, it just... breaks. And requires a reload for any page on the site to function. Looking at the console, it's a type error of some kind.
> 
> image
> 
> If you could fix this I'll merge it because it looks good otherwise

Should work now

File: frontend/src/main/components/Review/ReviewTable.js [hunk 1]
```diff
@@ -47,8 +47,7 @@ export default function ReviewTable({ reviews, currentUser }) {
         {
             Header: 'comments',
             accessor: 'comments',
-        }
-        
+        }
```
Comment (tallyhawley) [inline frontend/src/main/components/Review/ReviewTable.js:50]: on line 12 of this file - the url should say `review` not `reviews`. currently this does not match the existing routing.

Comment (tallyhawley) [inline frontend/src/main/components/Review/ReviewTable.js:50]: also, your itemID is not being accessed correctly. I think this should be `accessor: 'itemId'`, with a lowercase `d`. 
image

File: frontend/src/main/pages/Review/ReviewCreatePage.js [hunk 2]
```diff
@@ -1,13 +1,51 @@
...
+     );
+
+  const { isSuccess } = mutation
+
+  const onSubmit = async (data) => {
+    mutation.mutate(data);
+  }
+
+  if (isSuccess) {
+    return <Navigate to="/reviews/list" />
```
Comment (tallyhawley) [inline frontend/src/main/pages/Review/ReviewCreatePage.js:39]: this should navigate to `/review/list`, not `/reviews/list`

File: frontend/src/main/pages/Review/ReviewEditPage.js [hunk 3]
```diff
@@ -1,14 +1,72 @@
...
+
+
+  const objectToAxiosPutParams = (review) => ({
+    url: "/api/reviews",
+    method: "PUT",
+    params: {
+      id: review.id,
+    },
+    data: {
+      itemID: review.itemID,
```
Comment (tallyhawley) [inline frontend/src/main/pages/Review/ReviewEditPage.js:32]: should be `itemId: review.itemId`. the only case where this should say `itemID` (with capital D) is on the column header. not when you're accessing the actual parameter.

File: frontend/src/main/pages/Review/ReviewEditPage.js [hunk 4]
```diff
@@ -1,14 +1,72 @@
...
+  );
+
+  const { isSuccess } = mutation
+
+  const onSubmit = async (data) => {
+    mutation.mutate(data);
+  }
+
+  if (isSuccess) {
+    return <Navigate to="/reviews/list" />
```
Comment (tallyhawley) [inline frontend/src/main/pages/Review/ReviewEditPage.js:58]: `/review/list`

File: frontend/src/main/pages/Review/ReviewCreatePage.js [hunk 5]
```diff
@@ -1,13 +1,51 @@
...
+import { useBackendMutation } from "main/utils/useBackend";
+import { toast } from "react-toastify";
 
 export default function ReviewCreatePage() {
+
+  const objectToAxiosParams = (review) => ({
+    url: "/api/reviews/post",
+    method: "POST",
+    params: {
+      itemID: review.itemID,
```
Comment (tallyhawley) [inline frontend/src/main/pages/Review/ReviewCreatePage.js:13]: `itemId: review.itemId`

this is causing issues with communicating between frontend and backend.

**Review by tallyhawley — COMMENTED** [review]
looks mostly good. please fix the URL routing errors & the item ID access issues

**Comment by ethanlee556** [comment]
image
The itemID shows up now

**Review by tallyhawley — CHANGES_REQUESTED** [review]
could you update the tests so that they pass? should just be related to the itemId & /review/ vs /reviews/ changes

**Comment by ethanlee556** [comment]
Mutation Test Coverage:
image
image

**Review by tallyhawley — APPROVED** [review]
lgtm!

**Pull request merged**